File: OLMS.BackEnd/OLMS.BackEnd.API/Models/Resource.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using OLMS.BackEnd.Model;

namespace OLMS.BackEnd.API.Models
{
    public class Resource:Entity
    {
     
        [Index]
        [MaxLength(256)]
        [Required]
        public string Name { get; set; }
        [MaxLength(30)]
        [Index]
        [Required]
        public string Type { get; set; }
        [Index]
        [Required]
        public bool IsPublic { get; set; }
    }
}

File: OLMS.ClientApp/src/app/model/resource.ts
import { Entity } from './entity';

export class Resource extends Entity {
   
    name: string;
    type: string;
    isPublic: boolean;
}

export class ResourceType {
    constructor(id: string, value: string) {
        this.id = id;
        this.value = value;
    }

    id: string;
    value: string;
}